devicetypes/erocm123/inovelli-2-channel-smart-plug-mcd.src/inovelli-2-channel-smart-plug-mcd.groovy

```diff
@@ -0,0 +1,291 @@
+	sendHubCommand(cmds, 1000)
+}
+void childOff(String dni) {
+	logging("childOff($dni)", 1)
+	def cmds = []
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.basicV1.basicSet(value: 0x00), channelNumber(dni))))
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+	sendHubCommand(cmds, 1000)
+}
+void childRefresh(String dni) {
+	logging("childRefresh($dni)", 1)
+	def cmds = []
+	cmds << new physicalgraph.device.HubAction(command(encap(zwave.switchBinaryV1.switchBinaryGet(), channelNumber(dni))))
+	sendHubCommand(cmds, 1000)
+}
+def poll() {
+	logging("poll()", 1)
+	commands([
+			encap(zwave.switchBinaryV1.switchBinaryGet(), 1),
+			encap(zwave.switchBinaryV1.switchBinaryGet(), 2),
+	])
+}
+def refresh() {
+	logging("refresh()", 1)
+	commands([
+			encap(zwave.switchBinaryV1.switchBinaryGet(), 1),
+			encap(zwave.switchBinaryV1.switchBinaryGet(), 2),
+	])
+}
+def ping() {
+	logging("ping()", 1)
+	refresh()
+}
+def installed() {
+	logging("installed()", 1)
+	command(zwave.manufacturerSpecificV1.manufacturerSpecificGet())
+	createChildDevices()
+}
+def updated() {
+	logging("updated()", 1)
+	if (!childDevices) {
+		createChildDevices()
+	} else if (device.label != state.oldLabel) {
+		childDevices.each {
+			if (it.label == "${state.oldLabel} (CH${channelNumber(it.deviceNetworkId)})") {
+				def newLabel = "${device.displayName} (CH${channelNumber(it.deviceNetworkId)})"
+				it.setLabel(newLabel)
+			}
+		}
+		state.oldLabel = device.label
+	}
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID, offlinePingable: "1"])
+	sendEvent(name: "needUpdate", value: device.currentValue("needUpdate"), displayed: false, isStateChange: true)
+}
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	logging("${device.displayName} parameter '${cmd.parameterNumber}' with a byte size of '${cmd.size}' is set to '${cmd2Integer(cmd.configurationValue)}'", 2)
+}
+private encap(cmd, endpoint) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpoint).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+private command(physicalgraph.zwave.Command cmd) {
+	if (state.sec) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+private commands(commands, delay = 1000) {
+	delayBetween(commands.collect {
+		command(it)
+	}, delay)
+}
+private channelNumber(String dni) {
+	dni.split("-ep")[-1] as Integer
+}
+private void createChildDevices() {
+	state.oldLabel = device.label
+	for (i in 1..2) {
+		addChildDevice("Switch Child Device",
+				"${device.deviceNetworkId}-ep${i}",
+				device.hubId,
+				[completedSetup: true,
+				 label: "${device.displayName} (CH${i})",
+				 isComponent: true,
+				 componentName: "ep$i",
```

Comment:
These component names show up in the device details view in the app. I suggest changing the name to `outlet$i` which is more user friendly and matches the convention we use in other MCD DTHs (ex: button1, switch1, etc..)